File: app/src/main/java/com/magdenbt/collectionsbenchmark/UI/ViewFlow/MainActivity.java
package com.magdenbt.collectionsbenchmark.UI.ViewFlow;

import androidx.fragment.app.FragmentActivity;

import android.os.Bundle;

import com.magdenbt.collectionsbenchmark.InitApp;
import com.magdenbt.collectionsbenchmark.R;
import com.google.android.material.tabs.TabLayoutMediator;
import com.magdenbt.collectionsbenchmark.UI.SharedCollSizeVM;
import com.magdenbt.collectionsbenchmark.databinding.ActivityMainBinding;
import com.magdenbt.collectionsbenchmark.di.Qualifiers.MainActivityQ;

import javax.inject.Inject;

import dagger.Lazy;

public class MainActivity extends FragmentActivity {
    public static final String COLL_SIZE_REQ_KEY = "collestionSize";
    @Inject public Lazy<DialogCollectionSize> dialogCollectionSize;
    @Inject public VPAdapter vpAdapter;

    @MainActivityQ
    @Inject public SharedCollSizeVM sharedCollSizeVM;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        ((InitApp) getApplication()).getAppComponent()
                .mainComponent()
                .create(this)
                .inject(this);
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        getSupportFragmentManager().setFragmentResultListener(COLL_SIZE_REQ_KEY,this,(requestKey, result) -> sharedCollSizeVM.setCollectionsSize(result.getInt(COLL_SIZE_REQ_KEY)));
        if (sharedCollSizeVM.getCollectionSize() == 0) {
            dialogCollectionSize.get().setCancelable(false);
            dialogCollectionSize.get().show(getSupportFragmentManager(), "SomeTag");
        }

        binding.pager.setAdapter(vpAdapter);
        new TabLayoutMediator(binding.tab, binding.pager,false, (tab, position)->{
           switch (position){
               case 0:
                   tab.setText(getText(R.string.tab_collections));

                   break;
               case 1:
                   tab.setText(getText(R.string.tab_maps));
                   break;
           }
        }).attach();
    }

}

File: app/src/main/java/com/magdenbt/collectionsbenchmark/ui/viewflow/MainActivity.kt
package com.magdenbt.collectionsbenchmark.ui.viewflow

import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.get
import com.magdenbt.collectionsbenchmark.InitApp
import com.magdenbt.collectionsbenchmark.ui.SharedCollSizeViewModel
import com.magdenbt.collectionsbenchmark.ui.theme.AppTheme
import com.magdenbt.collectionsbenchmark.ui.viewmodelflow.StatViewModel
import com.magdenbt.collectionsbenchmark.ui.viewmodelflow.StatViewModelFactory
import javax.inject.Inject

class MainActivity : FragmentActivity() {

    @Inject
    lateinit var dialogCollectionSize: dagger.Lazy<DialogCollectionSize>

    @Inject
    lateinit var sharedCollSizeVM: SharedCollSizeViewModel

    companion object {
        const val COLL_SIZE_REQ_KEY = "collestionSize"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        (application as InitApp).appComponent.mainComponent().create(this).inject(this)
        super.onCreate(savedInstanceState)

        val viewModel: StatViewModel =
            ViewModelProvider(this, StatViewModelFactory(application)).get()

        supportFragmentManager.setFragmentResultListener(COLL_SIZE_REQ_KEY, this) { _, result ->
            sharedCollSizeVM.collectionSize.value = result.getInt(
                COLL_SIZE_REQ_KEY,
            )
        }

        if (sharedCollSizeVM.collectionSize.value == 0) {
            dialogCollectionSize.get().isCancelable = false
            dialogCollectionSize.get().show(supportFragmentManager, "SomeTag")
        }
        setContent {
            AppTheme {
                PagerScreen(viewModel = viewModel, collectionSize = sharedCollSizeVM.collectionSize)
            }
        }
    }
}
